front/lib/api/assistant/conversation.ts

```diff
@@ -1588,10 +1598,25 @@ async function* streamRunAgentEvents(
         yield event;
         break;
 
+      case "agent_generation_cancelled":
+        // Update status in database (and content from what was temporary saved in Redis ¯\_(ツ)_/¯ )
+        if (agentMessageRow.status !== "cancelled") {
+          const content = await getTemporaryMessageTokens(agentMessage.sId);
+          await agentMessageRow.update({
+            status: "cancelled",
+            content: content,
+          });
+          yield event;
+        }
+        break;
+
       // All other events that won't impact the database and are related to actions or tokens
       // generation.
       case "retrieval_params":
+        yield event;
+        break;
       case "generation_tokens":
+        await saveTemporaryMessageTokens(agentMessage.sId, event.text);
```

Comment:
why go to redis? you're in the same function between `saveTemporaryMessageTokens` and `getTemporaryMessageTokens`, no?